system/manager/build.py

```diff
@@ -1,55 +1,69 @@
 #!/usr/bin/env python3
 import os
 import subprocess
+import select
 from pathlib import Path
+import pyray as rl
```

Comment:
we shouldn't have to change the API for the consumers of Text and Spinner. how about we move the old wrappers to just use a thread instead of a subprocess and move them inside the text and spinner py files?